### PR TITLE
Disable previous drawing marks

### LINKED ISSUE #4467: Classifier: (previous) marks should be non-interactive when not on Drawing Task
## UI Bug

Package: `lib-classifier`
Related: #4456, which documents general cursor behaviour on Drawing Tasks.
Confirmed on 9e25158

On multi-task workflows, when a volunteer is on a _non-Drawing Task,_ then _previous_ drawn mark should be non-interactive.

- Expected: drawing annotations (marks) from Drawing Tasks from _previous Steps_ should still be visible on the Subject, but should not be interactive. (This is the behaviour on the PFE Classifier.) This means:
  - mouse/pointer cursor shouldn't change when on the mark
  - click/drag/etc events on the mark should be ignored
  - the mark should not be able to receive keyboard focus
- Actual: drawing annotations (marks) from Drawing Tasks from _previous Steps_ can have keyboard focus. Moving the mouse/pointer over the mark will show a "grab" cursor, though clicking/dragging still doesn't do anything.

### Testing

Confirmed with macOS + Chrome on a Macbook with a touchpad. (No touchscreen)

Setup:
- Create a simple Workflow with two tasks - T1: Drawing Task with a Circle tool, and T2: Text Task

Testing Steps:
- Open the Workflow in the Classifier.
- In T1, make a Circle mark, then proceed to T2
- In T2, move the mouse/pointer over the Circle mark created previously. Observe what happens.

### Status

Medium priority? Tagging @seanmiller26

### PR DESCRIPTION
**Pull request by eatyourgreens**
- Add an optional `disabled` prop to drawing tool marks.
- Disable marks from previous steps and tasks.
- Only apply interactive cursor styling to enabled marks.
- Remove `PreviousMarksConnector`.
- Test `PreviousMarks` with React Testing Library.

https://github.com/zooniverse/front-end-monorepo/assets/59547/3d736ebb-8d6a-45ce-b0fa-e8f9ffc3ad7c

_Please request review from `@zooniverse/frontend` team or an individual member of that team._ 

## Package
lib-classifier

## Linked Issue and/or Talk Post
- closes #4467.

## How to Review
On a drawing workflow with more than one step, marks from previous steps should now be disabled. Disabled marks are:
- not focusable.
- not styled as interactive.
- announced as disabled in a screen reader.

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [x] Tests are passing locally and on Github
- [ ] Documentation is up to date and changelog has been updated if appropriate
- [x] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [ ] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)
- [ ] FEM works in a mobile browser

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
- [ ] All pages of a FEM project load: Home Page, Classify Page, and About Pages
- [ ] Can submit a classification
- [ ] Can sign-in and sign-out
- [ ] The component is accessible
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/assistive-technology/testing-steps/voiceover-mac.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook

## Bug Fix
- [x] The PR creator has listed user actions to use when testing if bug is fixed
- [ ] The bug is fixed
- [ ] Unit tests are added or updated

### CODE REVIEW

**Comment by goplayoutside3** [comment]
Which project did you use to review locally?

**Comment by eatyourgreens** [comment]
Davy Notebooks is a good example (see #4467.)

File: packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/InteractionLayer/components/PreviousMarks/PreviousMarks.spec.js [hunk 1]
```diff
@@ -1,96 +1,182 @@
...
+        }]
+      },
+      // inactive drawing task
+      'T2': {
+        type: 'drawing',
+        tools: [{
+          type: 'point'
+        }]
+      },
+      // inactive drawing task
```
Comment (eatyourgreens) [inline packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/InteractionLayer/components/PreviousMarks/PreviousMarks.spec.js:30]:
```suggestion
      // inactive transcription task
```

**Review by goplayoutside3 — APPROVED** [review]
Looks good!

One NB question: When I use keyboard tab to move focus through the classifier, and a drawn mark on Davy Notebooks is disabled because I'm on the 2nd step of the task, I can still focus on the mark with my keyboard (at least I think it's focusing?). The component looks like this:
Screenshot 2023-06-22 at 3 15 56 PM

This PR adds a unit test to Mark.spec that says "it should not be focusable" so just wanted to double check if the above is expected behavior?

**Pull request merged**